.gitignore

```diff
@@ -9,3 +9,4 @@ cmake-build-release/
 lib/.DS_Store
 lib/TPP/.DS_Store
 compile_commands.json
+*.swp
```

Comment:
You can use `set directory` to put all your swap files under the same directory (say `~/.cache/vim`). :)